### PR TITLE
fix: correct posthog implementation

### LINKED ISSUE #1440: Bug: Posthog not sending analytics data
### Describe the bug

Open sauced will not send analytics data to posthog once deployed.

### Steps to reproduce

Run the application, click on the login button and then click on both onboarding butons in the onboarding flow.

### Affected services

opensauced.pizza

### Platforms

_No response_

### Browsers

_No response_

### Environment

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by 0-vortex**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR fixes a couple of issues with some previously fast-forwarded PRs:
- adjusts `jest` configuration to match hot.opensauced except typescript (still `esbuild`)
- adds correct `posthog` environment variable to `vite`
- aligns usage of `.env` files to the `vite` docs and hot.opensauced
- correctly mocks browser `fetch` function
- corrects social card to point to opensauced.pizza domain rather than github
- removes some `babel` constraints as part of the `jest` configuration (soon gone entirely)
- removes a rogue `console.log`
- switches `posthog` usage to non `dev` only

## Related Tickets & Documents

fixes #1440 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
Any concerns on putting the POSTHOG_ID in the .env. At the moment we are not leveraging the authorized URLs, but can.

Screen Shot 2022-05-14 at 6 25 20 AM

**Comment by 0-vortex** [comment]
> Any concerns on putting the POSTHOG_ID in the .env. At the moment we are not leveraging the authorized URLs, but can.
> 
> Screen Shot 2022-05-14 at 6 25 20 AM
> 
> Also would this make this PR not needed #1441

As with any js analytics suite (looking at google here), this is absolutely needed 🍕 The client variable being injected in the client code we need to have it public and protect the domains we take analytics from!

**Comment by 0-vortex** [comment]
Now it's ready 🍕

**Comment by chadstewart** [comment]
LGTM. My only concern is that do we need to push the .env file with the posthog id into the repo since it's already in declared in netlify? Just my only concern.

**Comment by 0-vortex** [comment]
> LGTM. My only concern is that do we need to push the .env file with the posthog id into the repo since it's already in declared in netlify? Just my only concern.

The static build will always contain the token, the code is executed on the browser. Making it available in the `.env` file is just a measure to support a future dev/staging environment - that way cloning the repo would let you use both local and dev/staging, while the production environment could change the token. However, this is just like a google analytics id, so, as long as we contain the domains the events can come from we're all set (posthog cloud) - we would not be able to secure that token by hiding or removing it from the `.env` file, just make collaboration more difficult.

**Comment by chadstewart** [comment]
> The static build will always contain the token, the code is executed on the browser. Making it available in the `.env` file is just a measure to support a future dev/staging environment - that way cloning the repo would let you use both local and dev/staging, while the production environment could change the token. However, this is just like a google analytics id, so, as long as we contain the domains the events can come from we're all set (posthog cloud) - we would not be able to secure that token by hiding or removing it from the `.env` file, just make collaboration more difficult.
> 
> 

Ah okay, I see, no problem. Learned something new today! Thanks much!

File: src/lib/analytics.js [hunk 1]
```diff
@@ -1,25 +1,25 @@
 import posthog from "posthog-js";
 
-function initiatePostHog () {
-    posthog.init(process.env.POSTHOG_ID, { api_host: 'https://app.posthog.com' });
-    return;
-}
+const initiatePostHog = () =>
+  posthog.init(import.meta.env.VITE_POSTHOG_API_KEY, { api_host: "https://app.posthog.com" });
 
-function capturePostHogAnayltics ( analyticsTitle, analyticsProperty, analyticsValue) {
```
Comment (mtfoley) [inline src/lib/analytics.js:8]: function name is a typo

**Comment by mtfoley** [comment]
I can see the token getting written in the bundle for the posthog.init() call after running `npm run build`, using command `grep -E -o '.{0,10}phc_.{0,5}' build/static/*.js`

No errors in the console when running locally, tests pass. Good with me.

File: src/lib/analytics.js [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-    const propertyIsMissingOrIncorrectType = !(typeof analyticsTitle === "string" &&
-                                               typeof analyticsProperty == "string" &&
-                                               analyticsValue);
-    if(propertyIsMissingOrIncorrectType) throw new Error("Property is missing or is incorrect type.");
+const capturePostHogAnayltics = (analyticsTitle, analyticsProperty, analyticsValue) => {
```
Comment (bdougie) [inline src/lib/analytics.js:6]:
```suggestion
const capturePostHogAnalytics = (analyticsTitle, analyticsProperty, analyticsValue) => {
```

File: src/lib/analytics.js [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
+  return posthog.capture(analyticsTitle, analyticsObject);
+};
 
 export {
-    initiatePostHog,
-    capturePostHogAnayltics
-};
\ No newline at end of file
+  initiatePostHog,
+  capturePostHogAnayltics
```
Comment (bdougie) [inline src/lib/analytics.js:24]:
```suggestion
  capturePostHogAnalytics
```

File: src/lib/repoStatusCode.test.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ describe("Test: repoStatusCode()", () => {
   });
 
   test("bdougie/open-sauced should return 200", async() => {
-    const repoUrl = "bdougie/open-sauced";
+    const repoUrl = "open-sauced/open-sauced";
```
Comment (bdougie) [inline src/lib/repoStatusCode.test.js:17]: 👍🏾

**Comment by chadstewart** [comment]
23H

**Pull request merged**